client/components/Book.jsx

```diff
@@ -29,18 +29,31 @@ class Book extends React.Component {
     });
   }
 
-  handleChange(e) {
-    // TO BE FILLED OUT
+  handleGuestCountChange(ageGroup, count) {
+    const {adults, children, infants} = this.state.guests;
+    const totalGuests = adults + children;
+    if (this.state.guests[ageGroup] === 0 && count === -1) {
+      return;
+    }
+    if (totalGuests >= 4 && count === 1 && ageGroup !== 'infants') {
+      return;
+    }
+    if (infants >= 5 && ageGroup === 'infants' && count === 1) {
+      return;
+    }
+    const guests = {...this.state.guests};
+    guests[ageGroup] = this.state.guests[ageGroup] + count;
```

Comment:
Got it. I had to write the code for this situation this particular way because it was a work around for setting the state of a specific prop of a nested object inside of a state prop. More specifically, I wanted to individually set the state of the count for adults/children/infants, but those props were nested inside of an object in my "guests" prop. So I found a way online to use destructuring to achieve this. Not ideal and def not recommended for deeply nested props, but worked for my situation.